cpp/src/arima/batched_arima.hpp

```diff
@@ -87,16 +123,102 @@ void residual(cumlHandle& handle, double* d_y, int num_batches, int nobs, int p,
+ * @param[in]  nobs        Number of samples per time series
+ *                         (all series must be identical)
+ * @param[in]  p           Number of AR parameters
+ * @param[in]  d           Difference parameter
+ * @param[in]  q           Number of MA parameters
+ * @param[in]  d_mu        mu if d != 0. Shape: (d, num_batches) (device)
+ * @param[in]  d_ar        AR parameters. Shape: (p, num_batches) (device)
+ * @param[in]  d_ma        MA parameters. Shape: (q, num_batches) (device)
+ * @param[out] ic          Array where to write the information criteria
+ *                         Shape: (num_batches) (host)
+ */
+void aic(cumlHandle& handle, double* d_y, int num_batches, int nobs, int p,
+         int d, int q, double* d_mu, double* d_ar, double* d_ma, double* ic);
+
+/**
+ * Compute corrected Akaike information criterion.
+ *
+ * @param[in]  handle      cuML handle
+ * @param[in]  d_y         Series to fit: shape = (nobs, num_batches) and
+ *                         expects column major data layout. (device)
+ * @param[in]  num_batches Total number of batched time series
+ * @param[in]  nobs        Number of samples per time series
+ *                         (all series must be identical)
+ * @param[in]  p           Number of AR parameters
+ * @param[in]  d           Difference parameter
+ * @param[in]  q           Number of MA parameters
+ * @param[in]  d_mu        mu if d != 0. Shape: (d, num_batches) (device)
+ * @param[in]  d_ar        AR parameters. Shape: (p, num_batches) (device)
+ * @param[in]  d_ma        MA parameters. Shape: (q, num_batches) (device)
+ * @param[out] ic          Array where to write the information criteria
+ *                         Shape: (num_batches) (host)
+ */
+void aicc(cumlHandle& handle, double* d_y, int num_batches, int nobs, int p,
+          int d, int q, double* d_mu, double* d_ar, double* d_ma, double* ic);
+
+/**
+ * Compute Bayesian information criterion.
+ *
+ * @param[in]  handle      cuML handle
+ * @param[in]  d_y         Series to fit: shape = (nobs, num_batches) and
+ *                         expects column major data layout. (device)
+ * @param[in]  num_batches Total number of batched time series
+ * @param[in]  nobs        Number of samples per time series
+ *                         (all series must be identical)
+ * @param[in]  p           Number of AR parameters
+ * @param[in]  d           Difference parameter
+ * @param[in]  q           Number of MA parameters
+ * @param[in]  d_mu        mu if d != 0. Shape: (d, num_batches) (device)
+ * @param[in]  d_ar        AR parameters. Shape: (p, num_batches) (device)
+ * @param[in]  d_ma        MA parameters. Shape: (q, num_batches) (device)
+ * @param[out] ic          Array where to write the information criteria
+ *                         Shape: (num_batches) (host)
+ */
+void bic(cumlHandle& handle, double* d_y, int num_batches, int nobs, int p,
+         int d, int q, double* d_mu, double* d_ar, double* d_ma, double* ic);
```

Comment:
@JohnZed @cjnolet these methods can also be generic "metrics" that we can add to our ever-growing list in ml-prims. I agree that the current interface is for batched case, but it seems useful to expose both non-batched as well as batched prims for these. What do you folks think?

Reply:
@Nyrio we already have a list of metrics implemented inside ml-prims [here](https://github.com/rapidsai/cuml/tree/branch-0.11/cpp/src_prims/metrics). It'll be useful for others also if you can create a `batched` sub-folder in here and moved these there. If you think it's better handled in a separate PR, let me know.

Reply:
@teju85 these functions are currently calling the likelihood function with the given parameters. If I understand well, you would like to move the metrics inside the prims and have them take the likelihood and the number of parameters, and then in ARIMA we would still have wrappers around the prims that calculate the likelihood and call the prims?

Reply:
@teju85 I have moved the information criterion to the location that you said, and abstracted a bit to remove redundancy. Also the IC is now calculated on GPU as I tweaked `batched_kalman_filter` to output the loglikelihood either as host or device array, on-demand, so I can compute the IC on GPU for no additional copy.